sdk/core/core-http/lib/nodeFetchHttpClient.ts

```diff
@@ -72,7 +81,8 @@ export class NodeFetchHttpClient extends FetchHttpClient {
       }
 
       const agentOptions: http.AgentOptions | https.AgentOptions = {
-        keepAlive: httpRequest.keepAlive
+        keepAlive: httpRequest.keepAlive,
```

Comment:
It looks like this change will only enforce the TLS version if either a proxy, or keepAlive are used. If neither are being used, then we're using the globalAgent and aren't setting a minimum version.

If we want to make this change, we might need to consider instantiating our own agent so we can set the minVersion/secureProtocol.

(This of course assumes we need this change at all :) )